lib/srv/db/mysql/test.go

```diff
@@ -0,0 +1,180 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mysql
+
+import (
+	"crypto/tls"
+	"io"
+	"net"
+	"strings"
+	"sync/atomic"
+
+	"github.com/gravitational/teleport/lib/srv/db/common"
+
+	"github.com/siddontang/go-mysql/client"
+	"github.com/siddontang/go-mysql/mysql"
+	"github.com/siddontang/go-mysql/server"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+// MakeTestClient returns MySQL client connection according to the provided
+// parameters.
+func MakeTestClient(config common.TestClientConfig) (*client.Conn, error) {
+	tlsConfig, err := common.MakeTestClientTLSConfig(config)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	conn, err := client.Connect(config.Address,
+		config.RouteToDatabase.Username,
+		"",
+		config.RouteToDatabase.Database,
+		func(conn *client.Conn) {
+			conn.SetTLSConfig(tlsConfig)
+		})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return conn, nil
+}
+
+// TestServer is a test MySQL server used in functional database
```

Comment:
I know it's potentially a lot of work but ideally this would use a real MySQL instance (e.g. running in a docker container) - just like etcd tests.

Reply:
Yeah I know. This is on my "long" list - I'd like to test against real Postgres/MySQL databases so we can test various protocol features as well. I have a [ticket](https://github.com/gravitational/teleport/issues/5025) about this, hopefully we'll get to it in not so distant future.